packages/moxb/src/progress/ProgressImpl.ts

```diff
@@ -0,0 +1,42 @@
+import { ValueImpl, ValueOptions } from '../value/ValueImpl';
+import { Progress } from './Progress';
+import { ValueOrFunction } from '../bind/BindImpl';
+import { computed } from 'mobx';
+
+export interface ProgressOptions extends ValueOptions<ProgressImpl, number> {
+    // The active status of the progress
+    status?: ValueOrFunction<'success' | 'exception' | 'normal' | 'active'>;
+    // Progress type
+    type?: ValueOrFunction<'line' | 'circle' | 'dashboard'>;
+    // Number of steps for line-type
+    steps?: ValueOrFunction<number>;
+}
+
+export class ProgressImpl extends ValueImpl<ProgressImpl, number, ProgressOptions> implements Progress {
+    constructor(impl: ProgressOptions) {
+        super(impl);
+    }
+
+    private implValOrUndefined(key: string) {
```

Comment:
Shouldn't this be moved into one of the base classes? (Like `ValueImpl` maybe?)

Reply:
Ok, moved to ValueImpl